firmware/defmt-test/src/lib.rs

```diff
@@ -8,6 +8,30 @@
 
 #![no_std]
 
+use defmt::Format;
 pub use defmt_test_macros::tests;
 
+/// Private implementation details used by the proc macro.
+#[doc(hidden)]
 pub mod export;
+
+/// Types indicating whether a test succeeded or failed.
+///
+/// This is comparable to the `Termination` trait in libstd, except stable and tailored towards the
+/// needs of defmt-test. It can be implemented by hand if desired, though normal usage shouldn't
+/// require that.
+pub trait TestOutcome: Format {
```

Comment:
unsure if we want to make this public right off the bat I would error on making it a sealed trait first.
if / when we want to make it public perhaps we should put the trait in the `defmt` crate because `defmt-test` depends on the `cortex-m` crate which is going to limit which crates can implement the trait.

Reply:
Yeah, sealed it for now